packages/fxa-auth-server/lib/routes/auth-schemes/mfa.ts

```diff
@@ -100,6 +100,10 @@ export const strategy = (
         throw AppError.invalidToken('Parent session token not found!');
       }
 
+      if (sessionToken.uid !== decoded.sub) {
```

Comment:
Nice! This seems like a good check to me. 

Can you put this in separate PR? We are trying to keep bug fixes / improvements that might be needed across multiple PRs in their own commits so in the event of a revert, we don't lose important changes.